lambda/src/main/scala/pricemigrationengine/handlers/CohortTableDatalakeExportHandler.scala

```diff
@@ -0,0 +1,158 @@
+      records <- CohortTable.fetchAll()
+      s3Location = S3Location(
+        s"price-migration-engine-${config.stage.toLowerCase}",
+        s"data.csv"
+      )
+      _ <- writeCsvToS3(records, s3Location, cohortSpec)
+    } yield HandlerOutput(isComplete = true)
+
+  def writeCsvToS3(
+      cohortItems: ZStream[Any, CohortFetchFailure, CohortItem],
+      s3Location: S3Location,
+      cohortSpec: CohortSpec
+  ): ZIO[S3 with Logging, Failure, Unit] =
+    localTempFile().use { filePath =>
+      for {
+        recordsWritten <- openOutputStream(filePath).use { tempFileOutputStream =>
+          writeCsvToStream(cohortItems, tempFileOutputStream, cohortSpec)
+        }
+
+        putResult <-
+          S3.putObject(s3Location, filePath.toFile)
+            .mapError { failure =>
+              CohortTableDatalakeExportFailure(s"Failed to write CohortItems to s3: ${failure.reason}")
+            }
+
+        _ <- Logging.info(
+          s"Successfully wrote cohort table ${cohortSpec.tmpTableName} containing $recordsWritten items " +
+            s"to $s3Location: $putResult"
+        )
+      } yield ()
+    }
+
+  def localTempFile(): ZManaged[Any, CohortTableDatalakeExportFailure, Path] =
+    ZManaged.make(
+      ZIO
+        .effect(Files.createTempFile(new File(TempFileDirectory).toPath, "CohortTableExport", "csv"))
+        .mapError { throwable =>
+          CohortTableDatalakeExportFailure(s"Failed to create temp file in $TempFileDirectory: ${throwable.getMessage}")
+        }
+    )(path => ZIO.effectTotal(Try(Files.delete(path))))
+
+  def openOutputStream(path: Path): ZManaged[Any, CohortTableDatalakeExportFailure, OutputStream] =
+    ZManaged.make(
+      ZIO
+        .effect(Files.newOutputStream(path))
+        .mapError { throwable =>
+          CohortTableDatalakeExportFailure(s"Failed to write to temp files $path: ${throwable.getMessage}")
+        }
+    )(stream => ZIO.effectTotal(stream.close()))
+
+  def writeCsvToStream(
+      cohortItems: ZStream[Any, Failure, CohortItem],
+      outputStream: OutputStream,
+      cohortSpec: CohortSpec
+  ): IO[Failure, Long] = {
+    managedCSVPrinter(
+      outputStream,
+      List(
+        "cohortName",
+        "cohortTable",
+        "subscriptionName",
+        "processingStage",
+        "startDate",
+        "currency",
+        "oldPrice",
+        "estimatedNewPrice",
+        "billingPeriod",
+        "whenEstimationDone",
+        "salesforcePriceRiseId",
+        "whenSfShowEstimate",
+        "newPrice",
+        "newSubscriptionId",
+        "whenAmendmentDone",
+        "whenNotificationSent",
+        "whenNotificationSentWrittenToSalesforce",
+        "whenAmendmentWrittenToSalesforce"
+      )
+    ).use { printer =>
+      cohortItems.mapM { cohortItem =>
+        ZIO
+          .effect(
+            printer.printRecord(
+              cohortSpec.cohortName,
+              cohortSpec.tmpTableName.getOrElse(""),
```

Comment:
Perhaps
```suggestion
              cohortSpec.tableName.getOrElse(""),
```
would be better to get table name in every case